src/coord/src/catalog.rs

```diff
@@ -1998,26 +1998,41 @@ impl Catalog {
     }
 
     /// Returns all tables, views, and sources in the same schemas as a set of
-    /// input ids.
-    pub fn schema_adjacent_relations(&self, sources: &[GlobalId], conn_id: u32) -> Vec<GlobalId> {
+    /// input ids. The indexes of all relations are included.
+    pub fn schema_adjacent_indexed_relations(
+        &self,
+        ids: &[GlobalId],
+        conn_id: u32,
+    ) -> Vec<GlobalId> {
         // Find all relations referenced by the expression. Find their parent schemas
         // and add all tables, views, and sources in those schemas to a set.
-        let mut ids = HashSet::new();
-        for id in sources {
+        let mut relations: HashSet<GlobalId> = HashSet::new();
+        let ids: HashSet<GlobalId> = ids.iter().copied().collect();
```

Comment:
did we construct `ids` the HashSet just to dedup the input array? wasn't sure if there was a more fundamental reason we have both `ids` and `relations`

Reply:
Oops, this was from a previous version. Removed.